### PR TITLE
Add support for apple silicon.

### PR DESCRIPTION
**Pull request by aarlt**
Depends on #14795.
Part of #12291.

### CODE REVIEW

File: .circleci/osx_install_dependencies.sh
Comment (nikola-matic) [inline .circleci/osx_install_dependencies.sh:1]: On line 66/67 you've got ` z3_dir="z3-${z3_version}-x64-osx-10.16"`. I'm guessing we'll need to swap this out with the arm version to fix the build - https://github.com/Z3Prover/z3/releases/tag/z3-4.12.1 (see 
z3-4.12.1-arm64-osx-11.0.zip).

Comment (nikola-matic) [reply .circleci/osx_install_dependencies.sh:1]: If you don't manage to get your CircleCI access up an running (so you can SSH), ping me, and I'll SSH into the session and try to manually download the ARM z3 distribution, and see if that fixes it.

Comment (aarlt) [reply .circleci/osx_install_dependencies.sh:1]: You're right! I didn't realised yesterday that z3 is not build from source :) Let me try this :)

File: .circleci/osx_install_dependencies.sh
Comment (nikola-matic) [inline .circleci/osx_install_dependencies.sh]: Are we planning on having 2 mac builds - both arm and x64? I would assume not, in which case, we don't really need this check.

Comment (aarlt) [reply .circleci/osx_install_dependencies.sh]: I think it's not really decided yet. However, maybe some users may use this script to setup the dependencies, some may still have x86.

Comment (nikola-matic) [reply .circleci/osx_install_dependencies.sh]: I really doubt anyone uses this to set up dependencies locally - but it does make sense to at least discuss keeping both builds for a while. E.g., I have an x64 Macbook, although I'm sure I'm in the minority :)

File: .circleci/osx_install_dependencies.sh
Comment (nikola-matic) [inline .circleci/osx_install_dependencies.sh:100]: Should probably `rm` this one after we're done with `make install`.

File: .circleci/osx_install_dependencies.sh
Comment (ekpyron) [reply .circleci/osx_install_dependencies.sh]: We said we'd look into how much hassle fat builds would be first...

Comment (ekpyron) [reply .circleci/osx_install_dependencies.sh]: Not sure how osx deals with dynamic libraries there though.

Comment (nikola-matic) [reply .circleci/osx_install_dependencies.sh]: Fat builds?

Comment (ekpyron) [reply .circleci/osx_install_dependencies.sh]: universal binaries that work across platforms natively. That should in theory just be a flag to xcode.

Comment (ekpyron) [reply .circleci/osx_install_dependencies.sh]: Although it remains to be seen how much hassle it'd be in practice.

Comment (nikola-matic) [reply .circleci/osx_install_dependencies.sh]: Ah, didn't know they were referred to as 'fat' builds.

Comment (ekpyron) [reply .circleci/osx_install_dependencies.sh]: Yeah, these days I guess universal binaries is more common

Comment (aarlt) [reply .circleci/osx_install_dependencies.sh]: ok, this PR implements now the creation of universal binaries... however, what about testing? right now this only tests the apple silicon part of the universal binaries..

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `636ac9427f5fd120ea4893d382d789b030a9b900`:
```
Error: Trailing whitespace found:
.circleci/config.yml:376:              cp /usr/local/bin/circleci /tmp/circleci 
.circleci/config.yml:382:              sudo cp /tmp/circleci /usr/local/bin/circleci 

```
Please check that your changes are working as intended.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `369e66c6954e395408db8507c66347201da529b8`:
```
Error: Trailing whitespace found:
.circleci/config.yml:382:              sudo cp /tmp/circleci /usr/local/bin/circleci 

```
Please check that your changes are working as intended.

File: .circleci/config.yml
Comment (ekpyron) [inline .circleci/config.yml:1151]: That's quite cool - if this is all that's needed, then we don't even need a headache over https://github.com/orgs/Homebrew/discussions/412 and homebrew accepting universal binaries or not, since the builds there can just set it as they want from the outside like this

File: .circleci/config.yml
Comment (cameel) [inline .circleci/config.yml]: I think the cache keys should include the architecture (`{{ arch }}`).

Comment (cameel) [reply .circleci/config.yml]: Also, wouldn't it be simpler to just check for `/usr/local/lib/libz3.a` (and add a comment why) than bother in a separate flag file that gets cached and restored separately?

File: .circleci/config.yml
Comment (cameel) [inline .circleci/config.yml]: Looks like this is addressing #12925. Could you make the cache fix and any related changes into a separate PR so that we can review and merge it separately first from the rest of the PR?

File: .circleci/compare_bytecode_reports.sh
Comment (cameel) [inline .circleci/compare_bytecode_reports.sh]: How about `intel` as a suffix for this? Also `arm` for ARM-only, though it looks like for now we don't need that one.

```suggestion
    osx_intel
```

File: .circleci/osx_install_dependencies.sh
Comment (cameel) [inline .circleci/osx_install_dependencies.sh]: You should quote variables.

File: .circleci/osx_install_dependencies.sh
Comment (cameel) [inline .circleci/osx_install_dependencies.sh:101]:
```suggestion
    git clone --depth 1 --branch "v${evmone_version}" https://github.com/ethereum/evmone.git
    cd evmone
```

But wouldn't it be better to just wget a source package from a tagged release instead?

File: test/scripts/test_bytecodecompare_prepare_report.py
Comment (cameel) [inline test/scripts/test_bytecodecompare_prepare_report.py]:
```suggestion
from bytecodecompare.prepare_report import ExecutionArchitecture, CompilerInterface, FileReport, ContractReport
from bytecodecompare.prepare_report import SettingsPreset, SMTUse, Statistics
```

**Review by cameel — COMMENTED** [review]
Haven't had time for a full review yet, here's some initial comments.

File: .circleci/config.yml
Comment (aarlt) [reply .circleci/config.yml]: The main problem is here, that `/usr/local/lib/libz3.a` need to be restored from cache. Thats why I introduced that caching flag thing, to easily know whether the cache exists. If it exists, it will first remove all files in `/usr/local`. The problem is that circleci is creating a message per already existing file, that drastically slows down everything. However, I will move that thing now to its own PR.

Comment (aarlt) [reply .circleci/config.yml]: See https://github.com/ethereum/solidity/pull/14795

File: .circleci/config.yml
Comment (aarlt) [reply .circleci/config.yml]: See https://github.com/ethereum/solidity/pull/14795

File: .circleci/osx_install_dependencies.sh
Comment (aarlt) [reply .circleci/osx_install_dependencies.sh:101]: I think its basically the same. I would just leave it like it is now.

File: .circleci/config.yml
Comment (cameel) [reply .circleci/config.yml]: Ah, ok. This should really have more comments to explain things like this.

Comment (aarlt) [reply .circleci/config.yml]: Yeah, true! I agree 😅 I'll add some comments

File: .circleci/osx_install_dependencies.sh
Comment (nikola-matic) [reply .circleci/osx_install_dependencies.sh:101]: `evmone` doesn't release with ARM builds yet, which is why we have to build from source.

File: scripts/bytecodecompare/prepare_report.py
Comment (nikola-matic) [inline scripts/bytecodecompare/prepare_report.py:446]: Listing the actual options would be helpful. Also, not a big fan of `native` vs `native-x86-64` nomenclature, I'd rather if it was just `arm64` and `x86-64`.

File: .circleci/osx_install_dependencies.sh
Comment (nikola-matic) [inline .circleci/osx_install_dependencies.sh:72]: Does `brew install boost` only provide an x86 build, i.e. is that why we're building boost from source?

File: .circleci/compare_bytecode_reports.sh
Comment (nikola-matic) [reply .circleci/compare_bytecode_reports.sh]: Thought you were nitpicking when first saw the comment, but going through the PR, I actually agree now :) Especially with regards to `native` vs `native-x86-64`.

File: .circleci/config.yml
Comment (nikola-matic) [inline .circleci/config.yml:496]: I'm a bit confused - we have a universal binary that can be run on both ARM and x86 (Intel) architectures - but we're running everything on an ARM machine in the CI - which means we're running the same binary twice (for generating the bytecode reports), but calling one x86 - even though they're both the same binary (arm)? Aside from the `binary_type` and `label` fields, everything else is the same in `b_bytecode_osx_x86_64` and `b_bytecode_osx` - and those fields are irrelevant anyway, since if we want to actually do the bytecode comparison tests between the two architectures, we'd have to have a separate `b_osx_x86` job that would use the the x86 resource class.

Either that, or I'm heavily misunderstanding how this works.

**Review by nikola-matic — COMMENTED** [review]
I have some concerns regarding the bytecode report/compare run for the two different architectures, the rest of the comments are not really that important.

File: scripts/bytecodecompare/prepare_report.py
Comment (nikola-matic) [reply scripts/bytecodecompare/prepare_report.py:446]: Or even just `arm` and `x86`, who even uses 32 bit addressing now days.

File: .circleci/osx_install_dependencies.sh
Comment (aarlt) [reply .circleci/osx_install_dependencies.sh:72]: I removed the brew based installation of boost. The thing is that we need the boost libraries in universal binary format, thats why we compile boost from source (`architecture=arm+x86`).

File: .circleci/config.yml
Comment (aarlt) [reply .circleci/config.yml:496]: We build universal binaries on that arm machine. Where we run all the tests for `arm` on `arm`. For the `x86_64` part we use the [`arch` command](https://www.unix.com/man-page/osx/1/arch/) `arch -64 <executable> <parameters>` to execute the `x86_64` portion of the universal binary to generate the bytecode reports.

Comment (aarlt) [reply .circleci/config.yml:496]: Oh! I just saw that I didn't passed the `arch_name` parameter :D

Comment (aarlt) [reply .circleci/config.yml:496]: Interesting, there seem to be two different intel architectures:
- `x86_64`   64-bit intel
- `x86_64h`  64-bit intel (haswell)

So I guess we need to compare the byte-code of both

File: .circleci/compare_bytecode_reports.sh
Comment (aarlt) [reply .circleci/compare_bytecode_reports.sh]: Turned out that there is also `x86_64h`, so I would say its the best to leave it like that. But I renamed it in `scripts/bytecodecompare/prepare_report.py` to just be `x86-64` and `x86-64h` - maybe the `native` prefix is a bit confusing there. In circleci config and in `compare_bytecode_reports.sh` script I just left the osx prefix.

File: scripts/bytecodecompare/prepare_report.py
Comment (aarlt) [reply scripts/bytecodecompare/prepare_report.py:446]: not sure what you mean.. but I agree that `native` vs `native-<something-else>` is confusing.. so I just call that `x86-64` and `x86-64h` there

File: .circleci/config.yml
Comment (nikola-matic) [reply .circleci/config.yml:496]: OK, so yeah, you can run x86 binaries 'natively' on ARM chips either via Rosetta or `arch` - of course, in the case of universal binaries, you have to explicitly state that you want to run the x86 binary.

Also, I'd ignore haswell and just use x86_64 intel (haswell is just a codename for a generation of intel's x86 chips).

**Comment by nikola-matic** [comment]
@aarlt is the dependency cache fix a per-requisite for this, or can we merge it after the release?

File: scripts/bytecodecompare/prepare_report.py
Comment (nikola-matic) [reply scripts/bytecodecompare/prepare_report.py:446]: Wait, `native` is just `arm` right? So `arm` and `x86` should be sufficient - no need for the `64`.

File: .circleci/config.yml
Comment (nikola-matic) [inline .circleci/config.yml]: Again, I'd remove the haswell job altogether.

File: scripts/bytecodecompare/prepare_report.py
Comment (nikola-matic) [inline scripts/bytecodecompare/prepare_report.py:235]: What exactly is `-64` here? Switching between the architectures is done via `arch -arm64` and `arch -x86_64` from what I can see.

Comment (aarlt) [reply scripts/bytecodecompare/prepare_report.py:235]: youre right! the arch name should be sufficient

File: .circleci/config.yml
Comment (aarlt) [reply .circleci/config.yml]: Why? I think it cannot hurt to have also these jobs.

**Comment by aarlt** [comment]
> @aarlt is the dependency cache fix a per-requisite for this, or can we merge it after the release?

Its not strictly needed, but it would be great to have - here we do a lot compilation from source (boost, z3 & evmone).. but without the cache we would need to do that very often, again and again.. I think we really need a working cache for that

File: .circleci/config.yml
Comment (aarlt) [reply .circleci/config.yml:496]: I added this, because the `arch` command seem to treat that slightly different as `x86_64` - they put haswell as a special architecture parameter `x86_64h`

File: scripts/bytecodecompare/prepare_report.py
Comment (aarlt) [reply scripts/bytecodecompare/prepare_report.py:446]: Yeah, I guess I could just change that

File: scripts/bytecodecompare/prepare_report.py
Comment (aarlt) [reply scripts/bytecodecompare/prepare_report.py:235]: turned out that only arch name is not sufficient:
```
COMPILER STDERR:
arch: posix_spawnp: /Users/distiller/project/build/solc/solc: Bad CPU type in executable
```

File: scripts/bytecodecompare/prepare_report.py
Comment (aarlt) [reply scripts/bytecodecompare/prepare_report.py:446]: there is now only `native` and `osx_intel`

File: .circleci/config.yml
Comment (aarlt) [reply .circleci/config.yml]: I removed that: we dont have any special `x86_64h` code inside our universal binary.

**Comment by nikola-matic** [comment]
Wait, how is this merge-able without approves? At least I don't see any approvals...

**Comment by aarlt** [comment]
> Wait, how is this merge-able without approves? At least I don't see any approvals...

hmm.. true..

**Comment by aarlt** [comment]
@nikola-matic ah.. its just because this one would only merge into the `osx-caching` branch

**Comment by nikola-matic** [comment]
> @nikola-matic ah.. its just because this one would only merge into the `osx-caching` branch

Ah OK, I thought both branches were targeting develop.

File: .circleci/osx_install_dependencies.sh
Comment (cameel) [inline .circleci/osx_install_dependencies.sh:72]: Does `b2` respect our `MAKEFLAGS`? If not, we should explicitly use `-j` here.

File: .circleci/osx_install_dependencies.sh
Comment (cameel) [reply .circleci/osx_install_dependencies.sh:101]: > `evmone` doesn't release with ARM builds yet, which is why we have to build from source.

But I said *source* package.

> I think its basically the same. I would just leave it like it is now.

Actually, one practical reason to switch to a package would be that we could then verify the package hash. Without package hash embedded in the script, the cache won't get invalidated when the package changes. We still have version here so it would only happen if tags get moved, but just saying, there *is* a difference. I think that building from packages is generally preferable to getting code via git.

File: .circleci/osx_install_dependencies.sh
Comment (cameel) [inline .circleci/osx_install_dependencies.sh]: We have `MAKEFLAGS` set in CI config for all jobs according to the number of cores so `--parallel` should not be necessary. I'm actually not sure how they interact so I'd be worried about `--parallel` overwriting it with a different value that may or may not be autodetected properly.

File: .circleci/osx_install_dependencies.sh
Comment (cameel) [inline .circleci/osx_install_dependencies.sh:108]:
```suggestion
    rm -r evmone
```

Better to avoid `--force` if you don't need it and with `rm` you rarely do.

File: scripts/bytecodecompare/prepare_report.py
Comment (cameel) [inline scripts/bytecodecompare/prepare_report.py]: Convert it to a proper enum value:
```suggestion
        ExecutionArchitecture(options.execution_arch),
```
argparse gives you a string and you have to convert it to actually get an enum. Then you don't have to unpack it with `value` on every use.

File: scripts/bytecodecompare/prepare_report.py
Comment (cameel) [inline scripts/bytecodecompare/prepare_report.py]: You should assert that it must be one or the other value, because if we add a new one, this code will have to be updated:
```python
else:
    assert execution_arch == ExecutionArchitecture.NATIVE.value
```

File: scripts/bytecodecompare/prepare_report.py
Comment (cameel) [inline scripts/bytecodecompare/prepare_report.py]: Let's keep those two in sync:
```suggestion
    OSX_INTEL = 'osx_intel'
```
Making it sound generic, like x86_64 is misleading anyway. `arch` is a macOS-only thing, isn't it?

**Review by cameel — COMMENTED** [review]
Needs just a few tweaks and it's good.

I have also some doubts about parallelism in the dependency installation script, but if you don't manage to get to those today, it would also be fine to deal with those in a separate PR, after the release.

**Comment by cameel** [comment]
Oh, an important question: did you test the binary outside of CI? Are we sure that people will be able to properly use it as a static binary when we release it tomorrow?

File: .circleci/config.yml
Comment (cameel) [inline .circleci/config.yml:1151]: There are quite a few warnings like these emitted [at the build step](https://app.circleci.com/pipelines/github/ethereum/solidity/32627/workflows/70749d51-5b9f-49b8-a54b-0e3ac566b16d/jobs/1461247):
```
ld: warning: ignoring duplicate libraries: '../libevmasm/libevmasm.a', '../libsmtutil/libsmtutil.a', '../libsolidity/libsolidity.a', '../libsolutil/libsolutil.a', '../libyul/libyul.a', '/usr/local/lib/libboost_filesystem.a'
ld: warning: ignoring duplicate libraries: '../libevmasm/libevmasm.a', '../libsmtutil/libsmtutil.a', '../libsolidity/libsolidity.a', '../libsolutil/libsolutil.a', '../libyul/libyul.a', '/usr/local/lib/libboost_filesystem.a'
```

What's happening there?

File: scripts/bytecodecompare/prepare_report.py
Comment (aarlt) [reply scripts/bytecodecompare/prepare_report.py]: thats great!

**Comment by cameel** [comment]
Looks like build takes slightly longer now (but could be a fluke, variability is quite large), but we got almost 2x speed up in tests and more than that in bytecode comparison.

[last run on `develop`](https://app.circleci.com/pipelines/github/ethereum/solidity/32627/workflows/70749d51-5b9f-49b8-a54b-0e3ac566b16d) vs [last run in this PR](https://app.circleci.com/pipelines/github/ethereum/solidity/32630/workflows/5f01de40-2ff5-4dde-880e-8e1bec00c6df)
| job                                       | `develop` | this PR |
|-------------------------------------------|-----------|---------|
| `b_osx`                                   | 10m 34s   | 12m 23s |
| `t_osx_cli`                               |  8m 46s   |  3m 49s |
| `t_osx_soltest`                           |  7m 46s   |  4m 24s |
| `b_bytecode_osx-via-ir-optimize`          | 10m 58s   |  4m 25s |
| `b_bytecode_osx_intel-via-ir-optimize`    |           |  4m 28s |
| `b_bytecode_osx-via-ir-no-optimize`       |  6m 9s    |  2m 29s |
| `b_bytecode_osx_intel-via-ir-no-optimize` |           |  2m 33s |
| `b_bytecode_osx-legacy-optimize`          |  3m 56s   |  1m 32s |
| `b_bytecode_osx_intel-legacy-optimize`    |           |  1m 43s |
| `b_bytecode_osx-legacy-no-optimize`       |  2m 12s   |  1m 1s  |
| `b_bytecode_osx_intel-legacy-no-optimize` |           |  1m 27s |

**Comment by aarlt** [comment]
> Oh, an important question: did you test the binary outside of CI? Are we sure that people will be able to properly use it as a static binary when we release it tomorrow?

I was not yet able to test the binary outside of the CI. I still have limited access to circleCI, maybe I should take that as an oportunity to setup my circle-ci account.

But I asked @nikola-matic yesterday to investigate the binaries, the defined architectures and used libaries. It looked good.

**Comment by cameel** [comment]
> I still have limited access to circleCI

You can get artifacts without logging in though. For example I can see [`solc` from the last run on the `Artifacts` page](https://app.circleci.com/pipelines/github/ethereum/solidity/32627/workflows/70749d51-5b9f-49b8-a54b-0e3ac566b16d/jobs/1461247/artifacts). Is it blank for you?

**Comment by aarlt** [comment]
> Looks like build takes slightly longer now (but could be a fluke, variability is quite large), but we got almost 2x speed up in tests and more than that in bytecode comparison.
> 

Thats great :)

I think it makes sense that the build is slower, because we are building now for two architectures.

**Comment by aarlt** [comment]
> > I still have limited access to circleCI
> 
> You can get artifacts without logging in though. For example I can see [`solc` from the last run on the `Artifacts` page](https://app.circleci.com/pipelines/github/ethereum/solidity/32627/workflows/70749d51-5b9f-49b8-a54b-0e3ac566b16d/jobs/1461247/artifacts). Is it blank for you?

Awesome! I didn't knew that! Thanks!

I just tested the binary locally and it looks good!

**Comment by cameel** [comment]
@nikola-matic If you can, please also try the binary in the morning before you start the release - as a final sanity check for this.

**Comment by nikola-matic** [comment]
> @nikola-matic If you can, please also try the binary in the morning before you start the release - as a final sanity check for this.

Sure thing - mine is x86_64, and we already know the ARM binary is good.

File: .circleci/config.yml
Comment (aarlt) [reply .circleci/config.yml:1151]: Hmm.. the truth is that I don't exactly know what is causing that. But I think it is not critical.

However, I see these warnings always when I run a build on macOS. This PR is not responsible for these warnings.  

```
[100%] Building CXX object test/CMakeFiles/soltest.dir/yulPhaser/SimulationRNG.cpp.o
[100%] Linking CXX executable soltest
ld: warning: ignoring duplicate libraries: '../libevmasm/libevmasm.a', '../libsmtutil/libsmtutil.a', '../libsolidity/libsolidity.a', '../libsolutil/libsolutil.a', '../libyul/libyul.a', '/opt/homebrew/lib/libboost_filesystem-mt.a'
[100%] Built target soltest
```

Not sure weather it makes sense to investigate this further. You don't see similar warnings using `gcc` on linux?

File: .circleci/osx_install_dependencies.sh
Comment (aarlt) [reply .circleci/osx_install_dependencies.sh:72]: It will probably not check the `MAKEFLAGS`. Looking at the `b2` documenation: "Default number of jobs is the number of detected available CPU threads. " But maybe its saver just to specify it, how much threads should I define here?

File: .circleci/config.yml
Comment (cameel) [reply .circleci/config.yml:1151]: No, I haven't seen them so far in linux builds. There also aren't any in the output from [`b_osx` run on `develop`](https://app.circleci.com/pipelines/github/ethereum/solidity/32627/workflows/70749d51-5b9f-49b8-a54b-0e3ac566b16d).

But ok, I guess if it was broken to the point of either not linking one of these libs or linking the one for the wrong platform, the binary would not work at all. So we probably don't have to worry about it. I think it's still worth trying to google around and see what may be causing it, but it's not critical. It's fine to check this after release.

File: .circleci/osx_install_dependencies.sh
Comment (aarlt) [reply .circleci/osx_install_dependencies.sh:101]: Ok, I can change it to use the source tar.

File: .circleci/osx_install_dependencies.sh
Comment (aarlt) [reply .circleci/osx_install_dependencies.sh:108]: if removed, wouldnt it ask for a confirmation by a user to delete the files?

Comment (aarlt) [reply .circleci/osx_install_dependencies.sh:108]: ah ok, looks like its ok

File: .circleci/osx_install_dependencies.sh
Comment (aarlt) [reply .circleci/osx_install_dependencies.sh]: I just removed `--parallel` here

File: .circleci/osx_install_dependencies.sh
Comment (aarlt) [reply .circleci/osx_install_dependencies.sh:72]: I would just leave it for now by that. I don't think it would create any problems not specifying a specific `-j` value.

Comment (cameel) [reply .circleci/osx_install_dependencies.sh:72]: If `-j` just works, then you could use that.

But if not, use the same value as in `MAKEFLAGS`. It would be best not to hard-code it here though, because it may differ between jobs. Using `$MAKEFLAGS` directly is also risky because it could contain stuff other than just `-j`. You could add a new variable next to it in CI config and put just the number in it. Then use that variable here. It also should have a default in the script - probably `1` unless there's some value that makes it auto-detect.

Comment (cameel) [reply .circleci/osx_install_dependencies.sh:72]: > I would just leave it for now by that. I don't think it would create any problems not specifying a specific `-j` value.

Yeah, if `-j` without a value works and makes it autodetect, it's fine too. I mean, now that I think about it, maybe we should generally use `-j` in CI. But that's something to consider later.

File: .circleci/osx_install_dependencies.sh
Comment (cameel) [reply .circleci/osx_install_dependencies.sh:108]: > if removed, wouldnt it ask for a confirmation by a user to delete the files?

No, it's the opposite. You actually have to use `--interactive` if you want it to ask.

`-f` is either for ignoring a file that does not exist, or ignoring permissions. Maybe also other cases, but generally ones that should not normally happen and if they do happen you at least want to take a look why before you decide that `-f` is warranted.

File: .circleci/osx_install_dependencies.sh
Comment (aarlt) [reply .circleci/osx_install_dependencies.sh:101]: ok.. the source package of evm one is not working 😅

They didn't packed the submodules inside.

```
CMake Error at CMakeLists.txt:8 (message):
  Git submodules not initialized, execute:

    git submodule update --init
```

Comment (cameel) [reply .circleci/osx_install_dependencies.sh:101]: Oh well... I guess github does not package submodules. It's fine then. Leave it as is.

File: .circleci/osx_install_dependencies.sh
Comment (aarlt) [reply .circleci/osx_install_dependencies.sh:72]: If `-j` is not set, `b2` will automatically determine the number of threads. So I think it's ok not to specify it.

Comment (cameel) [reply .circleci/osx_install_dependencies.sh:72]: OK. Please add a comment about this though :P This wasn't obvious to us at all.

**Review by cameel — DISMISSED** [review]
I'm approving since the only remaining things are not all that important (https://github.com/ethereum/solidity/pull/14725#discussion_r1465479799 and https://github.com/ethereum/solidity/pull/14725#discussion_r1465584452), but would be best if you still managed to address them before you merge.

**Review by cameel — APPROVED** [review]

**Pull request merged**
